torch/csrc/autograd/saved_variable.cpp

```diff
@@ -101,6 +102,13 @@ void SavedVariable::save_metadata(const Variable& data) {
     fw_grad_ = std::make_shared<ForwardGrad>();
     fw_grad_->set_value(fw_grad, /* level */ 0);
   }
+
+  original_metadata_ = std::make_shared<dynamo::autograd::VariableMetadata>(
```

Comment:
@albanD @soulitzer this change will require SavedVariable to always stash the pack hook input's metadata, this could be used in eager to enforce that unpack hook is returning a tensor with the same meta